### PR TITLE
 Dropdown options are shifted off of the dropdown list element. #389

### PR DESCRIPTION
**Pull request by erangeles**
…out of bounds, fixes shifting of page to the right. @notmessenger alex is going to add testing for this in issue 230 https://huboard.com/softlayer/sl-ember-components/#/issues/56565066 , please merge thanks.

### CODE REVIEW

File: addon/components/sl-select.js [hunk 1]
```diff
@@ -240,7 +240,7 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
                 query: ( query ) => {
                     const content = this.get( 'content' ) || [];
                     const optionLabelPath = this.get( 'optionLabelPath' );
-                    const select2 = this;
+                    const select2 = input.data( 'select2' ).opts;
```
Comment (notmessenger) [inline addon/components/sl-select.js]: What effect does this change have?  Is it related to https://github.com/softlayer/sl-ember-components/issues/433?

File: bower.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "fontawesome": "~4.3.0",
     "highcharts": "~4.1.5",
     "jquery": "^1.11.1",
+    "jquery-mousewheel": "^3.1.12",
```
Comment (notmessenger) [inline bower.json]: Why is this needed?

**Comment by notmessenger** [comment]
Setting aside whether or not the jquery-mousewheel dependency is need or not, do you not also need to make an addition to https://github.com/softlayer/sl-ember-components/blob/develop/blueprints/sl-ember-components/index.js?

**Comment by notmessenger** [comment]
How were you able to replicate this problem?  Visiting http://softlayer.github.io/sl-ember-components/#/demos/sl-select using Chrome 43.0.2357.130 (64-bit) on OS X 10.10.4 (14E46) I am unable to recreate it.

**Comment by erangeles** [comment]
@notmessenger are you looking at develop branch? this issue was introduced by https://github.com/softlayer/sl-ember-components/pull/404

File: addon/components/sl-select.js (hunk 1 above)
Comment (erangeles) [inline addon/components/sl-select.js]: this previously referenced the options scope of select2, this now references the view, matcher is a function in the options of select2.

File: bower.json (hunk 2 above)
Comment (erangeles) [inline bower.json]: the inclusion of jquery-mousewheel prevents scrolling outside the select menu, which is currently broken.

File: addon/components/sl-select.js [hunk 3]
```diff
@@ -267,16 +267,6 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
                 this.set( 'value', input.select2( 'val' ) );
             });
 
-            const originalBodyOverflow = document.body.style.overflow || 'auto';
```
Comment (erangeles) [inline addon/components/sl-select.js]: The removal of this block of code fixes the shifting issue, where the whole page would shift to one side. https://huboard.com/softlayer/sl-ember-components/#/issues/93135058 referenced in this issue.

**Comment by erangeles** [comment]
@notmessenger https://github.com/softlayer/sl-ember-components/blob/develop/blueprints/sl-ember-components/index.js I honestly don't know if this file has any bearing on injecting dependencies ( or what it really is to be honest). I have removed dependencies in the past from this file on purpose and built the project and it worked fine(to test what this file actually does). I will add the jquery-mousewheel dependency here if needed however I did not find a need for it as it worked fine with my current changes here.

**Comment by notmessenger** [comment]
Issue https://github.com/softlayer/sl-ember-components/issues/230 is no longer a valid issue, so any tests it was going to provide against these code changes needs to be incorporated into this PR.

File: bower.json (hunk 2 above)
Comment (notmessenger) [inline bower.json]: If https://github.com/softlayer/sl-ember-components/pull/434#discussion_r35028305 resolved bug https://github.com/softlayer/sl-ember-components/issues/433 then how is this change related or why is it needed?  Is it adding anything to the fix?  If it is something that has been determined as being desired for the functionality of this component then it should be done in a separate issue.
